lib/mailjet/mailer.rb

```diff
@@ -27,25 +27,187 @@ def initialize(options = {})
 # The deliver methods maps the Mail::Message attributes to the MailjetSend API JSON expected structure
 class Mailjet::APIMailer
   def initialize(options = {})
-    @delivery_method_options = options.slice(
+#    send = Mailjet.Send.new
+#    if send.version.exists
+#      version = send.version
+#    else
+      @version = Mailjet.config.api_version
+#    end
+    @delivery_method_options_v3_0 = options.slice(
       :recipients, :'mj-prio', :'mj-campaign', :'mj-deduplicatecampaign',
       :'mj-templatelanguage', :'mj-templateerrorreporting', :'mj-templateerrordeliver', :'mj-templateid',
       :'mj-trackopen', :'mj-trackclick',
-      :'mj-customid', :'mj-eventpayload', :vars, :headers
+      :'mj-customid', :'mj-eventpayload', :vars, :headers,
+    )
+    @delivery_method_options_v3_1 = options.slice(
+      :'Priority', :'CustomCampaign', :'DeduplicateCampaign',
+      :'TemplateLanguage', :'TemplateErrorReporting', :'TemplateErrorDeliver', :'TemplateID',
+      :'TrackOpens', :'TrackClicks',
+      :'CustomID', :'EventPayload', :vars, :Headers,
     )
   end
 
-  def deliver!(mail)
+  def deliver!(mail, options = nil)
+#    p mail.header.fields
+    
+    if (options && options.kind_of?(Object) && options['version'].present?)
```

Comment:
actually, it's useful since `options` can be nil
then `options['version']` should raise a `NoMethodFound` error, trying to call `:[]`
but I think it would be better to declare `def deliver!(mail, options = {})`
this way, you don't have to test `options`, because if `options` is given, it is expected to be a Hash, if not, it is an empty `Hash`, then you only need to test `if options['version']`